### PR TITLE
Bug: hidden_states shape mismatch in pd disaggregation when eagle3 is enabled

### PR DESCRIPTION
**Pull request by jianfei-wangg**
## Motivation

A bug is found when I enable eagle3 and pd disaggregation at the same time.
image

## Modifications

truncate hidden_states captured during eagle3 before p->d transfer

## Accuracy Tests

After fix, eagle3 configuration with 3,3,8 gets a reasonable accept rate:
image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by ShangmingCai** [comment]
Can you check https://github.com/sgl-project/sglang/pull/9976 ? We're gonna merge that PR soon, and I will prepare another PR to improve the first token acceptance rate.

**Comment by jianfei-wangg** [comment]
> Can you check #9976 ? We're gonna merge that PR soon, and I will prepare another PR to improve the first token acceptance rate.

It seems that we are solving the same problem and I check what should transfer when pd is enabled:
image
I test the accept rate on a deepseek-like model。It shows that transferring spec_info.hidden_states or logits_output.hidden_states have little influence on accpept rate。But，spec_info.hidden_states is more suitable to keep consistent behavior with non-pd scenario。
So I will close this pr, merge #9976 please.
